pkg/api/v1beta3/types.go

```diff
@@ -820,12 +820,12 @@ type NodeSpec struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Queried from cloud provider, if available.
-	HostIP string `json:"hostIP,omitempty"`
 	// NodePhase is the current lifecycle phase of the node.
 	Phase NodePhase `json:"phase,omitempty"`
 	// Conditions is an array of current node conditions.
 	Conditions []NodeCondition `json:"conditions,omitempty"`
+	// Queried from cloud provider, if available.
+	Addresses []NodeAddress `json:"addresses,omitempty"`
```

Comment:
Currently, those addresses are queried from cloudprovider, and just use Node.Name if not running cloudprovider.  I think it is reconstructable.  The PopulateIPs method does this.